audio_service/android/src/main/java/com/ryanheise/audioservice/AudioService.java

```diff
@@ -466,8 +457,8 @@ private Notification buildNotification() {
                 builder.setContentText(description.getSubtitle());
             if (description.getDescription() != null)
                 builder.setSubText(description.getDescription());
-            if (description.getIconBitmap() != null)
-                builder.setLargeIcon(description.getIconBitmap());
+            if (artBitmap != null)
```

Comment:
These two lines should be synchronized.

Reply:
I'm not sure why?

Reply:
The second line assumes `artBitmap` is still not null, but now it is possible for `setMetadata` to update this from another thread, e.g. from `setState`.

Reply:
I mean the current method may be triggered from `setState` while `setMetadata` may interrupt from another thread.

Reply:
After `setMetadata` is called it will call this method to update the notification anyways. I'm not sure it's worth locking on it

Reply:
I don't think it would be a major inconvenience or a significant performance hit to do this:

```java
synchronized (this) {
    if (artBitmap != null)
        ....
}
```

On the other hand, doing this makes it possible to understand the threadsafety of this class as a unit, independently of the rest of the system.